### ISSUE TITLE
Keycloak 25 / keycloakx - ServiceMonitor does not work with default port property

### ISSUE THREAD

**Issue by maximilian-krauss** [codecentric/helm-charts#796]
Heyho,
after the upgrade to Keycloak 25 and Chartversion 2.5.1 we are not getting any metrics via the Prometheus Operator.

We noticed that since Keycloak 25 metrics are being exposed only on Port 9000. We had to change `port` in the ServiceMonitor to `targetPort` in order to make it work. 

Unfortunately it is not possible to overwrite this property in this chart. But before sending a PullRequest to change it here I want to ask if someone is able to pull the metrics with the default chart configuration?

**Comment by Tim-Schwalbe** [comment]
No, I just have the same problem. I don't get any metrics and not the UP metric

**Comment by maximilian-krauss** [comment]
Hah, if it's not just us, then I will send a PR to fix it 🙂 

I changed it directly via kubectl in the ServiceMonitor after the helm deployment.

You need to replace `port` with `targetPort` and then it should work after a few minutes.

**Comment by Tim-Schwalbe** [comment]
It not working for me this way. I get the Up Metrics now, but somehow it's always 0, but keycloak started and all seems to work.

**Comment by Tim-Schwalbe** [comment]
okay I think I see the problem here. The service is not even listening to port http-internal aka 9000. So I had to extend the service-http.yaml 

but I am really asking myself how it could work for you @maximilian-krauss 

`  ports:
    - name: '{{ .Values.http.internalPort }}'
      port: 9000
      protocol: TCP
      targetPort: '{{ .Values.http.internalPort }}' `

**Comment by maximilian-krauss** [comment]
That's weird. Does it work with the port property on the service monitor after you added the port to the service? 

As far as I understood that's the difference between port and targetPort:
![CleanShot 2024-09-20 at 12 44 48@2x](https://github.com/user-attachments/assets/ebb98aea-a021-498b-b363-15e2de03b9b9)
from: https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#endpoint

By using targetPort in the service monitor it uses the pod to get metrics, port tries to use the service. 

I will also try on my setup which combination works, if it's the service its easy to add via the 'extraPorts' property

**Comment by Tim-Schwalbe** [comment]
Yes once I added the port to the service I worked. I am just tried with targetPort and it's also working :D
So both is fine I think I will go with the targetPort Version. Looks cleaner for me. I think I had also to fix my networkpolicies that's why it was not work in the first place.

**Comment by maximilian-krauss** [comment]
The additional port in the service alone did the trick. Thanks for pointing that out 👍

**Comment by maximilian-krauss** [comment]
Do not close - issue is still relevant until #798 has been merged

**Comment by maximilian-krauss** [comment]
Thats not completed. Its awaiting #814 to be merged

**Comment by stefangries** [comment]
#814 merged. Released with keycloakx-7.0.1.
